src/coreclr/jit/gentree.cpp

```diff
@@ -21851,8 +21919,8 @@ GenTree* Compiler::gtNewSimdCmpOpNode(
         {
             if (simdSize == 64)
             {
-                assert(IsBaselineVector512IsaSupportedDebugOnly());
-                intrinsic                = NI_AVX512F_CompareEqualMask;
+                assert(compIsaSupportedDebugOnly(InstructionSet_EVEX));
```

Comment:
nit: Most of these `compIsa...(InstructionSet_EVEX)` queries could be using the `canUseEvexEncoding` and `canUseEvexEncodingDebugOnly` helper APIs instead

Reply:
True. switched the checks.